packages/app-project/src/shared/components/CollectionsModal/components/SelectCollection/SelectCollection.js

```diff
@@ -13,7 +19,7 @@ function SelectCollection ({
   selected = {},
   userID = ''
 }) {
-  const [searchText, setSearchText] = useState('')
+  const searchText = useRef('')
```

Comment:
This is the fix for all your collections being shown when you type exactly 4 characters. It stops the component from rerendering immediately whenever `searchText` changes.